ClassLibraryProjects/GenericsHomeworkTests/NodeTests.cs

```diff
@@ -0,0 +1,133 @@
+﻿using GenericsHomework;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace GenericsHomeworkTests;
+
+[TestClass]
+public class NodeTests
+{
+    [TestMethod]
+    public void Node_TypeString_Pass()
+    {
+        string testValue = "Test Value";
+        Node<string> test = new(testValue);
+
+        Assert.AreEqual<string>(testValue, test.Value);
+    }
+
+    [TestMethod]
+    public void Node_TypeInt_Pass()
+    {
+        int testValue = 42;
+        Node<int> test = new(testValue);
+
+        Assert.AreEqual<int>(testValue, test.Value);
+    }
+
+    [TestMethod]
+    public void ToString_ValueToString_Pass()
+    {
+        int testValue = 42;
+        Node<int> test = new(testValue);
+
+        Assert.AreEqual<string>(testValue.ToString(), test.ToString());
+    }
+
+    [TestMethod]
+    public void Append_Appends_Pass()
+    {
+        Node<int> node1 = new(42);
+        Node<int> node2 = new(0);
+        Node<int> node3 = new(314);
+        node1.Append(node2);
+        node2.Append(node3);
+
+        Assert.AreEqual<Node<int>>(node1.Next, node2);
+        Assert.AreEqual<Node<int>>(node2.Next, node3);
+        Assert.AreEqual<Node<int>>(node3.Next, node1);
+
+    }
+
+    [TestMethod]
+    public void Clear_Clears_Pass()
+    {
+        Node<int> node1 = new(42);
+        Node<int> node2 = new(0);
+        Node<int> node3 = new(314);
+        node1.Append(node2);
+        node2.Append(node3);
+
+        node1.Clear();
+        Assert.AreEqual(node1, node1.Next);
+
+    }
+
+    [TestMethod]
+    public void Clear_CloseLoops_Pass()
+    {
+        Node<int> node1 = new(42);
+        Node<int> node2 = new(0);
+        Node<int> node3 = new(314);
+        node1.Append(node2);
+        node2.Append(node3);
+
+        node1.Clear();
+        Assert.AreEqual(node2, node2.Next);
+
+        //Preserve Node2 in the case we want to use it in or as another Node
+
+    }
+
+
+    [TestMethod]
+    public void Exists_ChecksValue_Pass()
+    {
+        Node<int> node1 = new(42);
+        Node<int> node2 = new(0);
+        Node<int> node3 = new(314);
+        node1.Append(node2);
+        node2.Append(node3);
+
+
+        Assert.IsTrue(node1.Exists(0));
+    }
+
+    public void Exists_ChecksOwnValue_Pass()
+    {
+        Node<int> node1 = new(42);
+        Node<int> node2 = new(0);
+        Node<int> node3 = new(314);
+        node1.Append(node2);
+        node2.Append(node3);
+
+        Assert.IsTrue(node1.Exists(42));
+    }
+
+    [TestMethod]
+    public void Exists_NonExistingValue_Fail()
+    {
+        Node<int> node1 = new(42);
+        Node<int> node2 = new(0);
+        Node<int> node3 = new(314);
+        node1.Append(node2);
+        node2.Append(node3);
+
+        Assert.IsFalse(node1.Exists(1337));
+
+    }
```

Comment:
```suggestion
    public void Exists_NonExistingValue_ReturnsFalse()
    {
        // Arrange
        Node<int> node = new(42);
        node.Append(int.MaxValue);
        
        // Assert
        Assert.IsFalse(node.Exists(0));
    }
```
An example of how to make the tests work again after accepting suggestions.